src/Game/zMusic.cpp

```diff
@@ -29,17 +121,91 @@
 // func_800A7924
 #pragma GLOBAL_ASM("asm/Game/zMusic.s", "volume_update__Ff")
 
-// func_800A7AA0
-#pragma GLOBAL_ASM("asm/Game/zMusic.s", "zMusicKill__Fv")
+//Stop all tracks and set them to null.
+void zMusicKill()
+{
+	for (int32 i = 0; i < 2; i++)
+	{
+		zMusicTrackInfo* track = &sMusicTrack[i];
+		if (track->snd_id != 0)
+		{
+			xSndStop(track->snd_id);
+			track->snd_id = 0;
+			track->loop = 0;
+			track->unk = 0;
+		}
+	}
+}
 
-// func_800A7B10
-#pragma GLOBAL_ASM("asm/Game/zMusic.s", "zMusicPause__Fv")
+//Pause music by stopping any non-null tracks.
+void zMusicPause()
+{
+    if (sMusicPaused == 0)
+    {
+  		for (int32 i = 0; i < 2; i++)
+  		{
+  			zMusicTrackInfo* track = &sMusicTrack[i];
+  			if (track->snd_id != 0)
+  			{
+  				xSndStop(track->snd_id);
+  			}
+  		}
+  		sMusicPaused = 1;
+  	}
+}
 
 // func_800A7B7C
 #pragma GLOBAL_ASM("asm/Game/zMusic.s", "zMusicUnpause__Fi")
 
+//Only thing that doesn't compile correctly is the function call to xSndPlay, in which a bunch of stuff is in the wrong order it looks like.
+/*
+void zMusicUnpause(int32 kill)
+{
+  	if (sMusicPaused != 0)
+  	{
+  		for (int32 i = 0; i < 2; i++)
+  		{
+  			zMusicTrackInfo* track = &sMusicTrack[i];
+  			uint32 prioBase = 0x800 * i;
+  			if (track->snd_id != 0)
+  			{
+  				if (kill != 0)
+  				{
+  					track->snd_id = 0;
+  				}
+  				else
+  				{
+  					uint32 res = xSndPlay(track->assetID, track->lastVol, unkFloat, 0xff, prioBase | (int)(-track->loop | track->loop) >> 0x1f & 0x8000U | 0x10000 | 0x20000, 0, SND_CAT_MUSIC, unkFloat);
+  					track->snd_id = res;
+  				}
+  			}
+  		}
+  		sMusicPaused = 0;
+  	}
+
+}*/
+
 // func_800A7C60
 #pragma GLOBAL_ASM("asm/Game/zMusic.s", "zMusicSetVolume__Fff")
 
-// func_800A7C90
-#pragma GLOBAL_ASM("asm/Game/zMusic.s", "zMusicReset__Fv")
+//Not sure what's wrong with this one. Doesn't match in the slightest.
+/*void zMusicSetVolume(float32 vol, float32 delay)
+{
+	if (delay <= minDelay) {
+    	volume.end = vol;
```

Comment:
Please try to always format every single `cpp` or `h` file that you've changed with `clang-format`. We have a [formatting style](https://github.com/pslehisl/bfbbdecomp/blob/master/.clang-format) which we are trying to stick to, to keep all code consistent.

Note that `clang-format` won't fix formatting on anything inside a comment, so you'll have to uncomment it, format it, and then re-comment it.